clients/src/main/java/org/apache/kafka/common/security/oauthbearer/OAuthBearerExtensionsValidatorCallback.java

```diff
@@ -0,0 +1,105 @@
+
+import javax.security.auth.callback.Callback;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A {@code Callback} for use by the {@code SaslServer} implementation when it
+ * needs to validate the SASL extensions for the OAUTHBEARER mechanism
+ * Callback handlers should use the {@link #validate(String)}
+ * method to communicate valid extensions back to the SASL server.
+ * Callback handlers should use the
+ * {@link #error(String, String)} method to communicate validation errors back to
+ * the SASL Server.
+ * As per RFC-7628 (https://tools.ietf.org/html/rfc7628#section-3.1), unknown extensions must be ignored by the server.
+ * The callback handler implementation should simply ignore unknown extensions,
+ * not calling {@link #error(String, String)} nor {@link #validate(String)}.
+ * Callback handlers should communicate other problems by raising an {@code IOException}.
+ * <p>
+ * The OAuth bearer token is provided in the callback for better context in extension validation.
+ * It is very important that token validation is done in its own {@link OAuthBearerValidatorCallback}
+ * irregardless of provided extensions, as they are inherently insecure.
+ */
+public class OAuthBearerExtensionsValidatorCallback implements Callback {
+    private final OAuthBearerToken token;
+    private final SaslExtensions inputExtensions;
+    private final Map<String, String> validatedExtensions = new HashMap<>();
+    private final Map<String, String> invalidExtensions = new HashMap<>();
+
+    public OAuthBearerExtensionsValidatorCallback(OAuthBearerToken token, SaslExtensions extensions) {
+        this.token = Objects.requireNonNull(token);
+        this.inputExtensions = Objects.requireNonNull(extensions);
+    }
+
+    /**
+     * @return {@link OAuthBearerToken} the OAuth bearer token of the client
+     */
+    public OAuthBearerToken token() {
+        return token;
+    }
+
+    /**
+     * @return {@link SaslExtensions} consisting of the unvalidated extension names and values that were sent by the client
+     */
+    public SaslExtensions inputExtensions() {
+        return inputExtensions;
+    }
+
+    /**
+     * @return an unmodifiable {@link Map} consisting of the validated and recognized by the server extension names and values
+     */
+    public Map<String, String> validatedExtensions() {
+        return Collections.unmodifiableMap(validatedExtensions);
+    }
+
+    /**
+     * @return An immutable {@link Map} consisting of the name->error messages of extensions which failed validation
+     */
+    public Map<String, String> invalidExtensions() {
+        return Collections.unmodifiableMap(invalidExtensions);
+    }
+
+    /**
+     * Validates a specific extension in the original {@code inputExtensions} map
+     * @param extensionName - the name of the extension which was validated
+     */
+    public void validate(String extensionName) {
```

Comment:
I think the name `validate` is confusing because it suggests that the actual validation is done by this method, while actually this method is simply marking the extension name valid. So perhaps just `valid` or something along those lines would be better. Javadoc also needs updating once the name is changed.